pkg/client/fake/client.go

```diff
@@ -79,28 +81,42 @@ func NewFakeClientWithScheme(clientScheme *runtime.Scheme, initObjs ...runtime.O
 }
 
 func (t versionedTracker) Create(gvr schema.GroupVersionResource, obj runtime.Object, ns string) error {
-	if accessor, err := meta.Accessor(obj); err == nil {
-		if accessor.GetResourceVersion() == "" {
-			accessor.SetResourceVersion("1")
-		}
-	} else {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
 		return err
 	}
+	if accessor.GetResourceVersion() != "" {
```

Comment:
I think a reasonable additional check on Create|Update would be:

```
	name := accessor.GetName()
	if len(name) == 0 {
		return fmt.Errorf("name is required")
	}
```

Reply:
Not objecting to the idea, but I think it would be better in a dedicated PR so we have both changes appear in the changelog

Reply:
@djzager did you want to open an issue to track that change separately?